lib/services/versions-api-service.js

```diff
@@ -1,50 +1,37 @@
-// Copyright 2015, EMC, Inc.
+// Copyright 2016, EMC, Inc.
 
 'use strict';
 
-var di = require('di'),
-    express = require('express');
+var di = require('di');
 
-module.exports = versionRouterFactory;
-
-di.annotate(versionRouterFactory, new di.Provide('Http.Api.Versions'));
-di.annotate(versionRouterFactory,
+module.exports = versionApiServiceFactory;
+di.annotate(versionApiServiceFactory, new di.Provide('Http.Services.Api.Versions'));
+di.annotate(versionApiServiceFactory,
     new di.Inject(
-        'Http.Services.RestApi',
+        'Logger',
         'ChildProcess',
-        '_',
-        'Logger'
+        '_'
     )
 );
+function versionApiServiceFactory(
+    Logger,
+    ChildProcess,
+    _
+) {
+    var logger = Logger.initialize(versionApiServiceFactory);
+
+    function VersionApiService() {
+    }
 
-function versionRouterFactory (rest, ChildProcess, _, Logger) {
-    var logger = Logger.initialize(versionRouterFactory);
-    var router = express.Router();
     /**
-     * @api {get} /api/1.1/versions GET /
-     * @apiDescription get versions of packages installed
-     * @apiName versions-get
-     * @apiGroup versions
-     * @apiSuccess {json} version list of all versions
-     * @apiSuccessExample {json} Success-Response:
-     *     HTTP/1.1 200 Ok
-     *     [
-     *         {
-     *             "package": "on-http",
-     *             "version": "1.0-19"
-     *         },
-     *         {
-     *             "package": "on-tftp",
-     *             "version": "1.0-19"
-     *         },
-     *     ]
-     *
-     * @apiErrorExample {json} Error-Response:
-     *     HTTP/1.1 501 Not Implemented
-     *
+     * Find the installed versions
+     * @param {Object} [req] HTTP request
+     * @param {Object} [res] HTTP response
+     * @return {Promise}
      */
+    VersionApiService.prototype.findVersion = function(req, res) {
 
-    router.get('/versions', rest(function (req, res) {
+        // Find installed packages
```

Comment:
Hi Joe,

Our goal was to get the middleware integrated and Versions was one of the simpler routes to implement so we went with that one.

From: Joseph Heck [mailto:notifications@github.com]
Sent: Friday, January 15, 2016 2:29 PM
To: RackHD/on-http
Cc: Abildgaard Jr, William
Subject: Re: [on-http] Inital support for V2.0 API with Swagger (#90)


In lib/services/versions-api-service.js<https://github.com/RackHD/on-http/pull/90#discussion_r49894420>:

>

> -    router.get('/versions', rest(function (req, res) {

> +        // Find installed packages

this was horrifically ghetto (and I wrote it) and makes us specific to debian (as well as VM based deployments) for talking about versions. Based on the other pieces I'm reading above, we'd be better supported by 'versions' being relevant to the versions of the API we support, rather than "what software versions are also on this machine". Is that what you were heading for? Or is this simply more an "here's an example of an API in both 1.1 and 2.0 w/ swagger"?

(it's terrible when you're embarrassed by code you wrote some time ago to solve a point solution :-) )

—
Reply to this email directly or view it on GitHub<https://github.com/RackHD/on-http/pull/90/files#r49894420>.